Source/Cypher/app/src/main/java/com/semaphore_soft/apps/cypher/ConnectionLobbyActivity.java

```diff
@@ -67,6 +90,41 @@ public void onCreate(Bundle savedInstanceState)
         {
             playerID = 0;
 
+            String ip = "";
+            try
+            {
+                for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en.hasMoreElements();)
+                {
+                    NetworkInterface ni = en.nextElement();
+                    for (Enumeration<InetAddress> addresses = ni.getInetAddresses(); addresses.hasMoreElements();)
+                    {
+                        InetAddress inetAddress = addresses.nextElement();
+                        // Limit IP addresses shown to IPv4
+                        if (!inetAddress.isLoopbackAddress() && inetAddress instanceof Inet4Address)
+                        {
+                            ip = inetAddress.getHostAddress();
+                            Log.i("Lobby", ip);
+                        }
+                    }
+                }
+            }
+            catch (SocketException ex)
+            {
+                Log.e("Lobby", ex.toString());
+            }
+            
+            mServiceIntent = new Intent(this, NetworkingService.class);
+            mServiceIntent.setData(Uri.parse(NetworkingService.SETUP_SERVER));
+            startService(mServiceIntent);
+
+            mServiceIntent.setData(Uri.parse(NetworkingService.WRITE_TO_CLIENT));
+            mServiceIntent.putExtra("message", "Hello, World!");
+            mServiceIntent.putExtra("index", 0);
+            startService(mServiceIntent);
+
+            TextView ipAddress = (TextView) findViewById(R.id.ip_address);
+            ipAddress.setText("Your IP Address is: " + ip);
+
```

Comment:
I put it in the connection lobby since that is when the service/threads are called and I didn't want the client to try to connect before the host has started accepting connections